software/06.Nest/README.md

```diff
@@ -9,131 +9,200 @@
 In this workshop, you'll use [Nest](https://nestjs.com/), a framework built on top of [Express](https://expressjs.com/) to create efficient and scalable apps. It's a very opinionated framework that provides everything you need and has built-in integration with many other frameworks (queues, type ORMs, validators, documentation, and much more!).  
 Nest fully supports TypeScript (in fact it's [built in TypeScript](https://github.com/nestjs/nest)).
 
-
-> If you aren't familiar with TypeScript, you can read [this file](https://github.com/PoCInnovation/Workshops/blob/master/software/06.Nest/src/step0/learn.md) to learn the basics.
-
 ## Step 0: Setup
 
 All the required information to start this workshop can be found in [SETUP.md](./SETUP.md)
 
 ## Step 1: Basics
 
-- In the `app.controller.ts` file, create a `@Get()` route `/article` (the route `hello` should help you to do so)
-- You also have to create the corresponding provider to return a test value.
+📑 Description:
+In this step, you will learn the fundamental concepts of NestJS and its architecture.
+
+
+📚 Documentation:
+
+> 📖 The documentation about [modules](https://docs.nestjs.com/modules) and [providers](https://docs.nestjs.com/providers) will teach you some basic concepts and architecture of Nest. In NestJS, **dependency injection** is primarily achieved through the use of decorators and the underlying inversion of control **(IoC)** container. Here's how it works:
+
+- `📦 Providers`: In NestJS, components that can be injected into other components are called providers. Providers are defined using the @Injectable() decorator. They are typically used to encapsulate business logic, interact with databases or external APIs, or provide utility functions.
+
+- `〽️ Injection`: To inject a provider into another component, such as a controller or another provider, you can use the constructor of the dependent component. NestJS relies on TypeScript's type system to infer the required dependencies.
+
+📌 Tasks:
```

Comment:
keep the same order as the template (description, tasks, documentation and validation)